lib/plugins/plugin/lib/utils.js

```diff
@@ -22,14 +22,20 @@ module.exports = {
     const serverlessYmlFilePath = path.join(servicePath, 'serverless.yml');
     const serverlessYamlFilePath = path.join(servicePath, 'serverless.yaml');
     const serverlessJsonFilePath = path.join(servicePath, 'serverless.json');
+    const serverlessJsFilePath = path.join(servicePath, 'serverless.js');
 
     return fileExists(serverlessYmlFilePath)
     .then(ymlExists => {
       if (!ymlExists) {
         return fileExists(serverlessYamlFilePath)
         .then(yamlExists => {
           if (!yamlExists) {
-            return serverlessJsonFilePath;
+            return fileExists(serverlessJsonFilePath).then((jsonExists) => {
+              if (jsonExists) {
+                return serverlessJsonFilePath;
+              }
+              return serverlessJsFilePath;
```

Comment:
Shouldn't we check here for the existence of the JS file too and reject with an error as last resort?
```js
return fileExists(serverlessJsFilePath).then(jsExists => {
  if (jsExists) {
    return serverlessJsFilePath;
  }
  return BbPromise.reject(new this.serverless.classes.Error('Could not find any serverless service definition file.'));
});
```

Reply:
Refactored this method a bit, looks much nicer now.